### PR TITLE
PR-2061 For SLNP Loaned message should update the supplier symbol and…

### PR DESCRIPTION
**Pull request by JanisSaldabols**
… replace volumes

### CODE REVIEW

File: service/grails-app/services/org/olf/rs/statemodel/events/EventISO18626IncomingAbstractService.groovy [hunk 1]
```diff
@@ -145,9 +145,10 @@ public abstract class EventISO18626IncomingAbstractService extends AbstractEvent
                         processedSuccessfully = false;
                         errorType = ERROR_TYPE_UNABLE_TO_FIND_REQUEST;
                     } else {
-                        if ((request.supplyingInstitutionSymbol == null ||
+                        if (((request.supplyingInstitutionSymbol == null ||
                                 request.getSupplyingInstitutionSymbol().contains("null") ||
-                                    request.getSupplyingInstitutionSymbol() == ":") &&
+                                    request.getSupplyingInstitutionSymbol() == ":") ||
+                                EventMessageRequestIndService.isSlnpRequesterStateModel(request)) &&
                                         eventData.header?.supplyingAgencyId?.agencyIdType != null &&
```
Comment (jakub-id) [inline service/grails-app/services/org/olf/rs/statemodel/events/EventISO18626IncomingAbstractService.groovy:152]: @JanisSaldabols This won't work. For SLNP we need to remove the supplying institution for Unfilled/Cancelled. So set it even if the new value is empty or null.

File: service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionPatronRequestISO18626StatusChangeService.groovy [hunk 2]
```diff
@@ -16,13 +17,19 @@ import org.olf.rs.statemodel.StateModel
...
+
     @Override
     String name() {
         return(ReasonForMessage.MESSAGE_REASON_STATUS_CHANGE)
     }
 
     @Override
     ActionResultDetails performAction(PatronRequest request, Object parameters, ActionResultDetails actionResultDetails) {
+        if (EventMessageRequestIndService.isSlnpRequesterStateModel(request) && parameters.statusInfo?.status == STATUS_UNFILLED) {
+            request.state = request.stateModel.initialState
```
Comment (jakub-id) [inline service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionPatronRequestISO18626StatusChangeService.groovy:30]: @JanisSaldabols did you test that this works? Is this simpler than doing it in`ActionPatronRequestISO18626StatusChangeService`?

File: service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionISO18626RequesterService.groovy [hunk 3]
```diff
@@ -39,9 +40,13 @@ public abstract class ActionISO18626RequesterService extends ActionISO18626Servi
...
         request.lastSequenceReceived = sequenceResult.sequence;
 
+        if (EventMessageRequestIndService.isSlnpRequesterStateModel(request)) {
+            Set<RequestVolume> volumes = new HashSet<>(request.volumes)
+            volumes.forEach {it -> request.removeFromVolumes(it)}
+        }
 
         // if parameters.deliveryInfo.itemId then we should stash the item id
-        if (parameters?.deliveryInfo) {
+        if (parameters?.deliveryInfo) { // TODO check if status is Loaned or CopyCompleted
```
Comment (jakub-id) [inline service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionISO18626RequesterService.groovy:49]: @JanisSaldabols itemId handling to reset volumes is missing below.

**Review by jakub-id — CHANGES_REQUESTED** [review]

File: service/grails-app/services/org/olf/rs/statemodel/events/EventISO18626IncomingAbstractService.groovy (hunk 1 above)
Comment (JanisSaldabols) [reply service/grails-app/services/org/olf/rs/statemodel/events/EventISO18626IncomingAbstractService.groovy:152]: Updated

File: service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionPatronRequestISO18626StatusChangeService.groovy (hunk 2 above)
Comment (JanisSaldabols) [reply service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionPatronRequestISO18626StatusChangeService.groovy:30]: This is ActionPatronRequestISO18626StatusChangeService class

File: service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionISO18626RequesterService.groovy (hunk 3 above)
Comment (JanisSaldabols) [reply service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionISO18626RequesterService.groovy:49]: It is below 
I just added that for SLNP we always remove existing volumes

Comment (jakub-id) [reply service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionISO18626RequesterService.groovy:49]: Ok, is this code only used for StatusChange or also when creating new or retrying request?

**Review by jakub-id — APPROVED** [review]

**Pull request merged**